### PR TITLE
Reworks Actor and updates references

### LINKED ISSUE #1009: Improve Actor and Network actor by removing unnecessary event listeners
## Summary

There's no need to add PreUpdate and OnUpdate listeners if they are not being used.

### PR DESCRIPTION
**Pull request by joaoburatto**
## Summary

Reworks Actor to remove unused UpdateEvents, updates how initialization works, how variables are displayed, ordering of elements and documents the system. Updates all references too.

### CODE REVIEW

File: Assets/Scripts/SS3D/Rendering/SS3D.Rendering.asmdef [hunk 1]
```diff
@@ -1,4 +1,18 @@
 {
-	"name": "SS3D.Rendering",
-	"references":[ "GUID:eb242eaf9f292954eb8c6fbd63c46deb" ]
-}
+    "name": "SS3D.Rendering",
+    "rootNamespace": "",
+    "references": [
+        "GUID:eb242eaf9f292954eb8c6fbd63c46deb",
```
Comment (stilnat) [inline Assets/Scripts/SS3D/Rendering/SS3D.Rendering.asmdef:5]: why was that needed to be changed ?

File: Assets/Scripts/SS3D/Core/Behaviours/Actor.cs [hunk 2]
```diff
@@ -1,187 +1,270 @@
...
-                eventService?.RemoveListener(eventHandle);
-            } 
+                eventService.RemoveListener(eventHandle);
+            }
         }
 
-        #endregion
-
-        #region EVENT_CALLS
-        private void OnPreUpdate(ref EventContext context, in LastPreUpdateEvent e) { HandlePreUpdate(e.DeltaTime); }
```
Comment (stilnat) [inline Assets/Scripts/SS3D/Core/Behaviours/Actor.cs:168]: why do these event calls are removed ? Can you detail that a bit ?

**Review by stilnat — CHANGES_REQUESTED** [review]
Looks good to me, just a few question, and then some play test.

File: Assets/Scripts/SS3D/Rendering/SS3D.Rendering.asmdef (hunk 1 above)
Comment (joaoburatto) [reply Assets/Scripts/SS3D/Rendering/SS3D.Rendering.asmdef:5]: Had to reference Coimbra's Assemblies

File: Assets/Scripts/SS3D/Core/Behaviours/Actor.cs (hunk 2 above)
Comment (joaoburatto) [reply Assets/Scripts/SS3D/Core/Behaviours/Actor.cs:168]: The automatic event call subscription was removed due to it causing wasted performance when the method was not overriden, so all Actors would have OnUpdate being called all the time even if it has no logic occurring.

**Review by stilnat — CHANGES_REQUESTED** [review]
okay yeah just saw your answer. well that's clear. We just need to have solid doc about that, because many contributors might be very confused by doing things this way, it's not really obvious. Could you add in IActor, actor and network actor, an example to set up those events like you did in your PR for some classes ?

**Review by stilnat — APPROVED** [review]
Play test with client host and server going well, code changes are clear, it's ready.
